method not found in `AtomicBool`

(Related to #4 and #17.)
Using `shared-bus  v0.2.0`  I have  several examples building on several device HALs  but with  `stm32f0xx`  I have a problem  with `.compare_exchange`. (Also happening with git shared-bus  #0f429435.)

``` 
Compiling shared-bus v0.2.2 (https://github.com/Rahix/shared-bus#0f429435)
error[E0599]: no method named `compare_exchange` found for struct `AtomicBool` in the current scope
   --> /home/paul/.cargo/git/checkouts/shared-bus-afeeec54cdff0f33/0f42943/src/mutex.rs:208:14
    |
208 |             .compare_exchange(
    |              ^^^^^^^^^^^^^^^^ method not found in `AtomicBool`
```
CI with examples working on other device HALs is at https://github.com/pdgilbert/rust-integration-testing/runs/4998981227?check_suite_focus=true. (`stm32f030` is a job at the bottom.)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,6 +22,7 @@ once_cell = { version = "1.4.0", optional = true }
 cortex-m = { version = "0.6.3", optional = true }
 xtensa-lx = { version = "0.6.0", optional = true }
 spin = { version = "0.9.2", optional = true }
+atomic-polyfill = "0.1.6"
 
 [dev-dependencies]
 embedded-hal-mock = "0.8"

--- a/src/mutex.rs
+++ b/src/mutex.rs
@@ -184,7 +184,7 @@ mod tests {
 #[derive(Debug)]
 pub struct AtomicCheckMutex<BUS> {
     bus: core::cell::UnsafeCell<BUS>,
-    busy: core::sync::atomic::AtomicBool,
+    busy: atomic_polyfill::AtomicBool,
 }
 
 // It is explicitly safe to share this across threads because there is a coherency check using an
@@ -199,7 +199,7 @@ impl<BUS> BusMutex for AtomicCheckMutex<BUS> {
     fn create(v: BUS) -> Self {
         Self {
             bus: core::cell::UnsafeCell::new(v),
-            busy: core::sync::atomic::AtomicBool::from(false),
+            busy: atomic_polyfill::AtomicBool::from(false),
         }
     }
 

